### PR TITLE
Member object

### PR DESCRIPTION
**Pull request by MarzaElise**
Tried my best to make the member object. I feel like this is missing a lot of stuff, so feel free to comment and tell me everywhere i went wrong :)
- [API docs](https://discord.com/developers/docs/resources/guild#guild-member-object)

### CODE REVIEW

**Comment by MarzaElise** [comment]
also i suck at docs so haven't tested them either

**Comment by MarzaElise** [comment]
no changes are tested :vibe:

File: acord/models/member.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    Represents a guild member.
+
+    Attributes
+    ----------
+    user: :class:`acord.models.User`
+          The User object of the member. Not included in MESSAGE_CREATE and MESSAGE_UPDATE events
+    nick: :class:`str`
+          Guild specific nickname of the member
+    avatar: :class:`str`
+          Member\'s guild avatar hash
```
Comment (Seniatical) [inline acord/models/member.py:21]: redundant backslash

File: acord/models/member.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    roles: :class:`List[acord.models.Snowflake]`
```
Comment (Seniatical) [inline acord/models/member.py:22]: Should be `List[:class:\`Snowflake\`]`

File: acord/models/member.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    permissions: :class:`str`
+          Total permissions of the member in the channel. Including overwrites.
+    """
+    # Please check the doc strings before merging @Seniatical
+    # and tell me if i need to add anything else :)
+
+    con: Any # connection object
+
+    user: Optional[User] # not included in MESSAGE_CREATE and MESSAGE_UPDATE events
+
```
Comment (Seniatical) [inline acord/models/member.py:43]: May need to add on additional vars depending on how we receive the event

File: acord/models/member.py (hunk 1 above)
Comment (MarzaElise) [reply acord/models/member.py:21]: okithen

File: acord/models/member.py (hunk 3 above)
Comment (MarzaElise) [reply acord/models/member.py:43]: wdym "additional vars"?

Comment (Seniatical) [reply acord/models/member.py:43]: guild user belongs to
tho I think that's usually received with `GUILD_CREATE`

Comment (MarzaElise) [reply acord/models/member.py:43]: why isnt it documented if its included in the guild-member object- :moyai: they should have documented it and tell which event its sent in

Comment (MarzaElise) [reply acord/models/member.py:43]: I could do that if i get proper docs links :smugdoggo:

Comment (Seniatical) [reply acord/models/member.py:43]: > why isnt it documented if its included in the guild-member object- 🗿 they should have documented it and tell which event its sent in

Eh not too sure, you receive members from `GUILD_CREATE` and `GUILD_MEMBER_ADD`

Comment (Seniatical) [reply acord/models/member.py:43]: the param is called `guild_id` if im not wrong

Comment (Seniatical) [reply acord/models/member.py:43]: > the param is called `guild_id` if im not wrong

Yes it is.
https://discord.com/developers/docs/topics/gateway#guild-member-add

**Comment by Seniatical** [comment]
Merging PR to stop delaying the release of threads.

**Pull request merged**